frontend/src/components/SearchBar.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react'
+import { toast } from 'react-toastify';
+const config = require("../config/constants.json")
+
+export default class SearchBar extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = { address: '', location: {}, store: []}
+        
+    }
+
+    handleChange = (event) => {
+        this.setState({ address: event.target.value });
+    }
+    searchStore = () => {
+        let location = {
+            latitude: this.state.location.lat,
+            longitude: this.state.location.lng,
+        }
+        fetch(`${config.endpoint}/api/stores/getClosest`, {
+            method: "POST",
+            body: JSON.stringify(location),
+            dataType: "JSON",
+            headers: {
+                "Content-Type": "application/json; charset=utf-8",
+            }
+        }).then((resp) => {
+            return resp.json()
+        }).then((data) => {
+
+            if (data.status === "OK") {
+                this.setState(
+                    {
+                        store: data
+                    }
+                );
+                this.props.onFindStore(data, location)
+                toast.success("Esperamos que disfrutes tu compra.");
+            } else {
+                toast.info("No pudimos encontrar tienda cerca a tudirección, pero puedes intentar con otra");
+            }
+
+        }).catch((error) => {
+            toast.error("Algo salio mal, por favor vuelve a intentarlo")
+        })
+    }
+    searchAddress = () => {
+        fetch(`${config.endpoint}/api/users/getLocation`, {
+            method: "POST",
+            body: JSON.stringify({
+                "address": this.state.address
+            }),
+            dataType: "JSON",
+            headers: {
+                "Content-Type": "application/json; charset=utf-8",
+            }
+        }).then((resp) => {
+            return resp.json()
+        }).then((data) => {
+            if (data.location.status === "OK") {
+                this.setState(
+                    {
+                        location: data.location
+                    }
+                );
+                localStorage.setItem("location", data.location);
```

Comment:
Why do you need to store this data on localStorage?

Reply:
it is not necessary, I put it there with the idea of ​​using it when the user refreshed the application, so that it does not enter it again, however later I changed my mind and forgot to delete it